wire-ios-request-strategy/Sources/E2EIdentity/E2EIManager.swift

```diff
@@ -0,0 +1,107 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireCoreCrypto
+
+public protocol E2EIManagerInterface {
+
+    /// Fetch acme directory for hyperlinks.
+    func loadACMEDirectory() async -> AcmeDirectory?
+
+    /// Get a nonce for creating an account.
+    func getACMENonce(endpoint: String) async -> String?
+
+    /// Create a new account.
+    func createNewAccount(prevNonce: String, createAccountEndpoint: String) async -> String?
+
+}
+
+public final class E2EIManager: E2EIManagerInterface {
+
+    private var apiProvider: APIProvider
+    private var context: NSManagedObjectContext?
+
+    public init(apiProvider: APIProvider, context: NSManagedObjectContext) {
+        self.apiProvider = apiProvider
+        self.context = context
+    }
+
+    public func loadACMEDirectory() async -> AcmeDirectory? {
+        var directory: WireCoreCrypto.AcmeDirectory?
+
+        guard let acmeAPI = apiProvider.acmeAPI(apiVersion: .v5),
```

Comment:
Is the ACME api actually versioned? It's not part of the Wire REST api

Reply:
It's not versioned